### PR TITLE
Plane: add support for guided via command int

### PR DESCRIPTION
**Pull request by WickedShell**
Allows plane to transition into guided and a point with a single packet, the mode change behavior is dependent on the mavlink command requesting it. Also allows guided waypoints to loiter ccw if requested.

I will be flight testing this tommorow.

### CODE REVIEW

**Comment by WickedShell** [comment]
Flight tested and it worked nicely. Quite nice to be able to swap loiter directions on the fly.

File: libraries/AP_Math/location.cpp [hunk 1]
```diff
@@ -259,6 +259,21 @@ bool location_sanitize(const struct Location &defaultLoc, struct Location &loc)
...
     }
+
+    // limit lat/lng to appropriate ranges
+    if (abs(loc.lat) > 90 * 1e7 || abs(loc.lng) > 180 * 1e7) {
+        loc.lat = defaultLoc.lat;
+        loc.lng = defaultLoc.lng;
+        has_changed = true;
+    }
+
+    if (abs(loc.alt) >= 0x7fffff) {
```
Comment (tridge) [inline libraries/AP_Math/location.cpp]: this won't do anything as loc.alt is 24 bit, so this if() is never true.
in our discussion on checking this I presumed the check would happen on the value in the command before it was in the Location structure. Once it is in the Location structure its too late to tell if its out of range

Comment (WickedShell) [inline libraries/AP_Math/location.cpp]: The original goal was to catch saturation/excessively large values, but you're right that its almost never going to catch anything as it wont saturate. Will remove. (I do check at the input for a float as well)

**Comment by tridge** [comment]
applied, thanks!

**Pull request merged**